python/iron/device/device.py

```diff
@@ -53,32 +55,31 @@ def op(self, op: TileOp):
                 raise ValueError("Cannot set operation more than once.")
             self._op = op
 
-    def __init__(self, cols: int, rows: int) -> None:
+    def __init__(self, device: AIEDevice) -> None:
         """Initialize a representation of a device.
 
         Args:
-            cols (int): Number of columns on the device
-            rows (int): Number of rows on the device.
+            device (AIEDevice): aie device
         """
-        self._cols = cols
-        self._rows = rows
+        self._device = device
         self._tiles: list[list[Device.__DeviceTile]] = []
 
         # Create all "physical" tiles belonging to the device at initialization to
         # ensure only one "physical" tile object is every created corresponding to the same
         # coordinates.
```

Comment:
Not directly related to changes introduced by this PR but it caught my eye: "ever" instead of "every"?